wgpu/backends/wgpu_native/_api.py

```diff
@@ -367,6 +367,21 @@ def request_adapter_sync(
         This is the implementation based on wgpu-native.
         """
         check_can_use_sync_variants()
+        # Similar to https://github.com/gfx-rs/wgpu?tab=readme-ov-file#environment-variables
+        # It seems that the environment variables are only respected in their
+        # testing environments maybe????
+        # In Dec 2024 we couldn't get the use of their environment variables to work
+        # This should only be used in testing environments and API users
+        # should beware
+        # We chose the variable name WGPUPY_WGPU_ADAPTER_NAME instead WGPU_ADAPTER_NAME
+        # to avoid a clash
+        if adapter_name := os.getenv(("WGPUPY_WGPU_ADAPTER_NAME")):
+            adapters = self.enumerate_adapters_sync()
+            adapters_llvm = [a for a in adapters if adapter_name in a.summary]
+            if not adapters_llvm:
+                raise ValueError(f"Adapter with name '{adapter_name}' not found.")
+            return adapters_llvm[0]
```

Comment:
We can also just move this into `_request_adapter` to avoid duplicating code. We're in the wgpu-native backend here, so we know we can use the sync version.

Reply:
honestly, how does this code work:

https://github.com/pygfx/wgpu-py/blob/ae7efc554d302f836203cfc82dd43b2d044952c8/wgpu/backends/wgpu_native/_api.py#L449

there is no `awaitable` at the time the function is created.... is there some magic in `ffi.Callback`

Reply:
Ok i made my little attempt, but it clearly doesn't work.

Can one of you have a look please. I would love to learn how to use async stuff.....